### ISSUE TITLE
SkinnedMesh.clone doesn't work properly, results in undefined skeleton property

### ISSUE THREAD

**Issue by SkaceKamen** [mrdoob/three.js#19419]
When cloning `SkinnedMesh`, the resulting `SkinnedMesh` instance doesn't have `skeleton` property defined.

I'm not sure if this is also a bug, but having a `SkinnedMesh` with undefined `skeleton` crashes the renderer with:
```
 Uncaught TypeError: Cannot read property 'frame' of undefined
```

at

https://github.com/mrdoob/three.js/blob/f177bce88ea3ff48818a30ff09d871133fcd245b/src/renderers/WebGLRenderer.js#L1343

##### Example
https://jsfiddle.net/pwonz6gd/

##### Three.js version

- [x] dev
- [x] r115

##### Browser

- [x] All of them

##### OS

- [x] All of them

#### Solution?

Copying the skeleton is not as easy as it sounds, one has to also reassign all the bones from the previous mesh to the new mesh.

My quick-fix for this issue was this:
```typescript
export const fixSkeleton = (sourceScene: Object3D, target: Object3D) => {
  t.traverse((i) => {
    if (i instanceof SkinnedMesh) {
      const original = findByName(sourceScene, i.name) as SkinnedMesh
      i.skeleton = original .skeleton.clone()
      i.skeleton.bones = i.skeleton.bones.map(
        (b) => findByName(target, b.name) as Bone
      )

      i.bindMatrix = source.bindMatrix.clone()
      i.bindMatrixInverse = source.bindMatrixInverse.clone()
    }
  })
}
```

I think something similar would have to be implemented in the `SkinnedMesh` copy method, just not as simple as this (relies on unique names of bones).

**Comment by Mugen87** [comment]
You have to use [SkeletonUtils.clone()](https://threejs.org/docs/index.html#examples/en/utils/SkeletonUtils.clone) when cloning skinned meshes. Can you please give it a try?

**Comment by SkaceKamen** [comment]
I didn't know that's a thing, I'll try that!

The thing is, I'm not directly cloning SkinnedMesh, it's just part of the scene I'm cloning as part of import process - I'd expect this to be automatic? Or at least warn user that they're cloning something that's not clonable directly?

**Comment by gkjohnson** [comment]
#17370 would address this but it seems there hasn't been a lot of movement or even comment on it. There are similar problems for a lot of other objects that retain references to other objects like `DirectionalLights`. `.clone` will likely not do what you expect especially when cloning a full scene. I'm still hoping it'll get fixed at some point.

**Comment by SkaceKamen** [comment]
The #17370 would probably speed up the process a bit, but it wouldn't solve this problem

The thing is, the clone method is already fixed, just hidden in `SkeletonUtils.clone` ... is there something stopping us from moving it from there to `SkinnedMesh.clone`?

https://github.com/mrdoob/three.js/blob/dev/examples/jsm/utils/SkeletonUtils.js#L544

**Comment by Mugen87** [comment]
> is there something stopping us from moving it from there to SkinnedMesh.clone?

Yes, this is explained here: https://github.com/mrdoob/three.js/pull/14494#issue-202138275

**Comment by gkjohnson** [comment]
> The #17370 would probably speed up the process a bit, but it wouldn't solve this problem

Sorry I didn't mean to imply it would immediately solve this problem if merged but it introduces a pattern that could be used to solve the issues raised in the PR @Mugen87 linked to.

**Comment by vallafederico** [comment]
Shouldn't the docs reflect this? 

At least in ENG ([here](https://threejs.org/docs/#api/en/objects/SkinnedMesh.clone)) I get the .clone() on skinnedMesh page and doesn't look like it's fixed or will be if I'm getting this right. Maybe point to "use SkeletonUtils.clone instead".

**Comment by Mugen87** [comment]
That's a good point! Let me update the docs.
